logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java

```diff
@@ -207,6 +212,139 @@ public void shouldNotDeleteOnCloseByDefault() {
     assertThat(files.length).isGreaterThan(0);
   }
 
+  @Test
+  public void shouldDeleteFromLogStream() {
+    // given
+    final int segmentSize = 1024 * 8;
+    final int remainingCapacity =
+        (segmentSize
+            - FsLogSegmentDescriptor.METADATA_LENGTH
+            - alignedLength(HEADER_BLOCK_LENGTH + 2 + 8)
+            - 1);
+    final int idxSize = (segmentSize - FsLogSegmentDescriptor.METADATA_LENGTH);
+    final LogStream logStream =
+        buildLogStream(
+            c -> c.logSegmentSize(segmentSize).readBlockSize(idxSize).indexBlockSize(idxSize));
+    logStream.openAppender().join();
+    closeables.manage(logStream);
+    final byte[] largeEvent = new byte[remainingCapacity];
+
+    // written from segment 0 4096 -> 8192, idx block address 4096
+    final long firstPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+    // written from segment 1 4096 -> 8192, but idx block address segment 0 - 8192
+    final long secondPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+    // written from segment 2 4096 -> 8192, but idx block address segment 1 - 8192
+    final long thirdPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+    // written from segment 3 4096 -> 8192, but idx block address segment 2 - 8192
+    final long fourthPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+
+    logStream.setCommitPosition(fourthPosition);
+    waitUntil(() -> logStream.getLogBlockIndex().getLastPosition() >= fourthPosition);
+
+    // when
+    logStream.delete(fourthPosition);
+
+    // then
+    assertThat(events(logStream).count()).isEqualTo(2);
```

Comment:
It is not clear why we delete only 2 of 4 segments and not 3. Please add a comment or change the test case.